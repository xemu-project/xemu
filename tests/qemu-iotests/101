#!/usr/bin/env bash
# group: rw quick
#
# Test short file I/O
#
# Copyright (C) 2014 Red Hat, Inc.
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#

# creator
owner=stefanha@redhat.com

seq=`basename $0`
echo "QA output created by $seq"

status=1	# failure is the default!

_cleanup()
{
	_cleanup_test_img
}
trap "_cleanup; exit \$status" 0 1 2 3 15

# get standard environment, filters and checks
. ./common.rc
. ./common.filter

_supported_fmt raw
_supported_proto file
_supported_os Linux


echo
echo "== creating short image file =="
dd if=/dev/zero of="$TEST_IMG" bs=1 count=320

echo
echo "== reading bytes beyond EOF gives zeroes =="
$QEMU_IO -c "read -P 0 0 512" "$TEST_IMG" | _filter_qemu_io


# success, all done
echo "*** done"
rm -f $seq.full
status=0
