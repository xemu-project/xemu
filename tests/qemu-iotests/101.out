QA output created by 101

== creating short image file ==
320+0 records in
320+0 records out

== reading bytes beyond EOF gives zeroes ==
read 512/512 bytes at offset 0
512 bytes, X ops; XX:XX:XX.X (XXX YYY/sec and XXX ops/sec)
*** done
